### PR TITLE
add global auth cookie

### PR DESCRIPTION
**Pull request by devadaviddd**

### CODE REVIEW

File: apps/web/src/constants/common.ts [hunk 1]
```diff
@@ -2,6 +2,7 @@ import { version } from '../core/version';
 
 export const APP_VERSION = version;
 export const DEV_MODE = process.env.NODE_ENV === 'development';
+export const AUTH_DOMAIN = process.env.NEXT_PUBLIC_AUTH_DOMAIN || 'tuturuuu.com'
```
Comment (vhpx) [inline apps/web/src/constants/common.ts:5]:
```suggestion
export const AUTH_COOKIE_DOMAIN = process.env.NEXT_PUBLIC_AUTH_COOKIE_DOMAIN
```

File: apps/web/src/pages/_app.tsx [hunk 2]
```diff
@@ -8,6 +8,7 @@ import Providers from '../components/common/Providers';
 import Analytics from '../components/common/Analytics';
 import usePersistLocale from '../hooks/usePersistLocale';
 import { Database } from '../types/database.types';
+import { AUTH_DOMAIN, DEV_MODE } from '../constants/common';
```
Comment (vhpx) [inline apps/web/src/pages/_app.tsx:11]:
```suggestion
import { AUTH_COOKIE_DOMAIN, DEV_MODE } from '../constants/common';
```

File: apps/web/src/pages/_app.tsx [hunk 3]
```diff
@@ -17,7 +18,17 @@ export default function Application({
...
+    createPagesBrowserClient<Database>({
+      cookieOptions: {
+        domain: DEV_MODE ? undefined : AUTH_DOMAIN,
+        maxAge: 100000000,
+        path: '/',
+        sameSite: 'lax',
+        secure: true,
+      },
+    })
+  );
```
Comment (vhpx) [inline apps/web/src/pages/_app.tsx:31]:
```suggestion
    createPagesBrowserClient<Database>(
    (DEV_MODE || !AUTH_COOKIE_DOMAIN)
    ? undefined
    : {
      cookieOptions: {
        domain: AUTH_COOKIE_DOMAIN,
        maxAge: 100000000,
        path: '/',
        sameSite: 'lax',
        secure: true,
      },
    })
  );
```

File: turbo.json [hunk 4]
```diff
@@ -6,14 +6,13 @@
     "NEXT_PUBLIC_SUPABASE_ANON_KEY",
     "NEXT_PUBLIC_SUPABASE_URL",
     "SUPABASE_SERVICE_KEY",
-
+    "NEXT_PUBLIC_AUTH_DOMAIN",
```
Comment (vhpx) [inline turbo.json:9]:
```suggestion
    "NEXT_PUBLIC_AUTH_COOKIE_DOMAIN",
```

**Review by vhpx — CHANGES_REQUESTED** [review]
Please take a look at the suggested changes. After it's resolved, I will merge this PR to main and staging so that we can test this feature out before rolling into production.

**Comment by devadaviddd** [comment]
> Please take a look at the suggested changes. After it's resolved, I will merge this PR to main and staging so that we can test this feature out before rolling into production.

Thank you for the feedback on my pull request. I have reviewed your suggested changes and implemented them in a new commit. I believe I have addressed your comments - please take a look at the updated pull request when you have a chance. Let me know if you have any other changes you would like me to make. I appreciate you taking the time to provide suggestions to improve my code.

**Pull request merged**